nix-index-wrapper.nix

```diff
@@ -8,10 +8,25 @@ runCommand "nix-index-with-db-${nix-index-unwrapped.version}"
   nativeBuildInputs = [ makeWrapper ];
   meta.mainProgram = "nix-locate";
 } ''
+  # Function that does the same as nixpkgs.lib.versionAtLeast,
+  # return true if the first string denotes a version equal to or newer than
+  # the second one.
+  function versionAtLeast() {
+    printf '%s\n' "$2" "$1" | sort --version-sort --check=quiet
+  }
+
   mkdir -p $out/share/cache/nix-index
   ln -s ${nix-index-database} $out/share/cache/nix-index/files
-  makeWrapper ${nix-index-unwrapped}/bin/nix-locate $out/bin/nix-locate \
-    --set XDG_CACHE_HOME $out/share/cache
+
+  version="$(${nix-index-unwrapped}/bin/nix-index --version | cut -d' ' -f2)"
+
+  if versionAtLeast "''${version}" "0.1.6"; then
+    makeWrapper ${nix-index-unwrapped}/bin/nix-locate $out/bin/nix-locate \
+      --set NIX_INDEX_DATABASE $out/share/cache
```

Comment:
I think we need to point to `$out/share/cache/nix-index` here.

Reply:
From the source code (https://github.com/nix-community/nix-index/blob/2628e658385c3c8de53ae0c68bc9d7381d76b8e2/src/bin/nix-locate.rs#L242) I deduced that we point to the toplevel dir containing the `nix-index` dir.

It works in this way in any case, I tested it locally.

Reply:
It does, indeed, not work:

```
❯ nix shell github:R-VdP/nix-index-database/nix_index_env_var
❯ nix-locate nix-locate
error: reading from the database at '/nix/store/7zpyhmrr1h5i77macc61rv2sc7p0408h-nix-index-with-db-0.1.6/share/cache/files' failed.
❯ NIX_INDEX_DATABASE=/nix/store/7zpyhmrr1h5i77macc61rv2sc7p0408h-nix-index-with-db-0.1.6/share/cache/nix-index /nix/store/gzwdnj0183s527d9gar4k4cvcjzmbg41-nix-index-0.1.6/bin/nix-locate nix-locate
nix-index.out                                         0 s /nix/store/0r2lyhsijmw6p0mqc7g2kk4rdjizwhvl-nix-index-0.1.5/bin/nix-locate
```

Reply:
Hmm, weird, I thought this worked on my machine. I'll check again tonight.

Reply:
Okay, fixed, and I did the same for comma too.